hax/helper/generate_sysconf.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+# Module for Hare to create motr and s3server environment files.
+
+from typing import Dict, List, NamedTuple, Optional
+from hax.util import repeat_if_fails, KVAdapter
+from hax.types import Fid, ObjT
+import logging
+import os
+import re
+from helper.exec import Executor, Program
+import simplejson
+import fileinput
+
+
+def get_node_name():
+    with open('/var/lib/hare/node-name') as f:
+        return f.readline().strip('\n')
+
+
+def reload_consul():
+    executor = Executor()
+    logging.debug('Reloading consul')
+    executor.run(Program([
+        'consul', 'reload'
+    ]))
+
+
+class Service(NamedTuple):
+    id: str
+    name: str
+    address: str
+    port: int
+    checks: List[Dict]
+
+
+class Generator:
+    """
+    Generates system configuration files for consul, motr and s3 services.
+    """
+
+    def __init__(self):
+        self.kv = KVAdapter()
+        self.sysconfig_dir = '/etc/sysconfig/'
+        if not os.path.isdir(self.sysconfig_dir):
+            self.sysconfig_dir = '/etc/default/'
+
+    @repeat_if_fails()
+    def get_service_ids(self, svc_name: str) -> List[str]:
```

Comment:
As discussed let's have offline version of these functions too.

Reply:
done, added 2 seperate classes to hold the online and offline versions. Based on kv file input, functions are called.